src/multipart.rs

```diff
@@ -88,6 +88,29 @@ pub(crate) struct MultipartState<'r> {
     pub(crate) constraints: Constraints,
 }
 
+#[derive(Debug)]
+pub(crate) struct Boundary(String);
+
+impl Boundary {
+    fn new(boundary: String) -> Self {
+        let cap = constants::CRLF.len() + constants::BOUNDARY_EXT.len() + boundary.len();
+        let mut buf = String::with_capacity(cap);
+        buf.push_str(constants::CRLF);
+        buf.push_str(constants::BOUNDARY_EXT);
+        buf.push_str(&boundary);
+        Self(buf)
```

Comment:
Though this might be better than using `format!()` directly, the hope would be to remove all unnecessary allocations along the parsing path. Instead of using `format!()` to create a string to compare, perform the comparison directly. That is, instead of:

```rust
"ab" == format!("{}{}", "a", "b");
```

Do:

```rust
"ab"[..1] == "a" && "ab"[1..] == "b"
```

But safely, of course, considering all bounds and possible string lengths.